frontend/src/lib/fabricCanvasUtils.ts

```diff
@@ -0,0 +1,364 @@
+  const jsonData = await response.json();
+  if (canvas) {
+    canvas.loadFromJSON(jsonData, () => canvas.renderAll());
+  }
+}
+
+export const handleAddText = (
+  canvas: fabric.Canvas | null,
+  posX: number,
+  posY: number,
+  options?: { text?: string; color?: string; fontSize?: number }
+): boolean => {
+  const { text = "Type here...", color = "#333", fontSize = 20 } = options || {};
+  
+  const newText = new fabric.IText(text, {
+    left: posX,
+    top: posY,
+    fontFamily: "arial",
+    fill: color,
+    fontSize: fontSize,
+    lockScalingY: true, 
+    lockScalingX: true,
+    lockSkewingX: true,
+    lockSkewingY: true,
+    lockScalingFlip: true, 
+  });
+ 
+  canvas?.add(newText);
+  canvas?.setActiveObject(newText);
+  newText.enterEditing();
+  newText.selectAll();
+
+  return false;
+};
+
+
+
+export const handleRemoveSelected = (canvas: fabric.Canvas | null): void => {
+  if (canvas) {
+    const activeObjects = canvas.getActiveObjects();
+    activeObjects.forEach((obj) => canvas.remove(obj));
+    canvas.discardActiveObject().requestRenderAll();
+  }
+};
+
+export const handleGroupSelected = (canvas: fabric.Canvas | null): void => {
+  if (canvas) {
+    const activeObjects = canvas.getActiveObjects();
+    if (activeObjects.length) {
+      const clonedObjects = activeObjects.map((obj) =>
+        fabric.util.object.clone(obj)
+      );
+
+      const group = new fabric.Group(clonedObjects, {
+        originX: "center",
+        originY: "center",
+      });
+
+      activeObjects.forEach((obj) => canvas.remove(obj));
+      canvas.add(group);
+      canvas.setActiveObject(group);
+      canvas.requestRenderAll();
+    }
+  }
+};
+
+export const addLine = (canvas: fabric.Canvas | null, options?: { color?: string; strokeWidth?: number }): boolean => {
+  const { color = "black", strokeWidth = 5 } = options || {};
+  const line = new fabric.Line([50, 100, 200, 200], {
+    stroke: color,
+    strokeWidth: strokeWidth,
+    selectable: true,
+  });
+  canvas?.add(line);
+  canvas?.setActiveObject(line);
+  return false;
+};
+
+export const addRectangle = (canvas: fabric.Canvas | null, options?: { fillColor?: string; width?: number; height?: number }): void => {
+  const { fillColor = "red", width = 200, height = 100 } = options || {};
+  const rect = new fabric.Rect({
+    left: 100,
+    top: 100,
+    fill: fillColor,
+    width: width,
+    height: height,
+    angle: 0,
+    transparentCorners: false,
+  });
+  canvas?.add(rect);
+  canvas?.setActiveObject(rect);
+};
+
+export const addCircle = (canvas: fabric.Canvas | null, options?: { fillColor?: string; radius?: number }): void => {
+  const { fillColor = "green", radius = 50 } = options || {};
+  const circle = new fabric.Circle({
+    radius: radius,
+    fill: fillColor,
+    left: 200,
+    top: 200,
+  });
+  canvas?.add(circle);
+  canvas?.setActiveObject(circle);
+};
+
+const removeImagesFromCanvas = (canvas:fabric.Canvas | null) => {
```

Comment:
It would be nice to indicate that we return something
```suggestion
/**
 * Removes all images from the canva
 * @param canvas The canva to remove images from
 * @returns Removed images in a form of imagesData
 */
const removeImagesFromCanvas = (canvas:fabric.Canvas | null) => {
```